09-clustering.Rmd

```diff
@@ -148,79 +152,83 @@ In K-means clustering, we measure the quality of a cluster by its
 First, we find the cluster centers by computing the mean of each variable 
 over data points in the cluster. For example, suppose we have a 
-cluster containing 3 observations, and we are using two variables, $x$ and $y$, to cluster the data.
+cluster containing four observations, and we are using two variables, $x$ and $y$, to cluster the data.
 Then we would compute the $x$ and $y$ variables, $\mu_x$ and $\mu_y$, of the cluster center via
 
-$$\mu_x = \frac{1}{3}(x_1+x_2+x_3) \quad \mu_y = \frac{1}{3}(y_1+y_2+y_3).$$
+$$\mu_x = \frac{1}{4}(x_1+x_2+x_3+x_4) \quad \mu_y = \frac{1}{4}(y_1+y_2+y_3+y_4).$$
 
-In the first cluster from the customer satisfaction/loyalty example, there 
-are 5 data points. These are shown with their cluster center 
-(`csat = 1.8` and `loyalty = 7.5`) highlighted below.
+In the first cluster from the example, there are `r nrow(clus1)` data points. These are shown with their cluster center 
+(``r paste("flipper_length_scaled =", round(mean(clus1$flipper_length_scaled),2))`` and ``r paste("bill_length_scaled =", round(mean(clus1$bill_length_scaled),2))``) highlighted in figure \@ref(fig:10-toy-example-clus1-center).
 
 ```{r 10-toy-example-clus1-center, echo = FALSE, warning = FALSE, fig.height = 4, fig.width = 4.35, fig.cap = "Cluster 1 from the toy example, with center highlighted."}
 base <- ggplot(clus1) +
-  geom_point(aes(y = loyalty, x = csat)) +
-  labs(x = "Customer satisfaction", y = "Loyalty") +
-  xlim(c(0, 4)) +
-  ylim(c(6, 9)) +
-  geom_point(aes(y = mean(loyalty), x = mean(csat), colour="red", size=2.5)) +
+  geom_point(aes(y = bill_length_scaled, x = flipper_length_scaled)) +
+  labs(x = "Flipper Length (scaled)", y = "Bill Length (scaled)") +
+  xlim(c(
+    min(clus1$flipper_length_scaled) - 0.25 * sd(clus1$flipper_length_scaled),
+    max(clus1$flipper_length_scaled) + 0.25 * sd(clus1$flipper_length_scaled)
+  )) +
+  ylim(c(
+    min(clus1$bill_length_scaled) - 0.25 * sd(clus1$bill_length_scaled),
+    max(clus1$bill_length_scaled) + 0.25 * sd(clus1$bill_length_scaled)
+  )) +
+  geom_point(aes(y = mean(bill_length_scaled), x = mean(flipper_length_scaled), colour = "red", size = 2.5)) +
   theme(legend.position = "none")
 base
 ```
 
 The second step in computing the WSSD is to add up the squared distance between each point in the cluster and the cluster center.
 We use the straight-line / Euclidean distance formula that we learned about in the classification chapter.
-In the 3-observation cluster example above, we would compute the WSSD $S^2$ via
+In the `r nrow(clus1)`-observation cluster example above, we would compute the WSSD $S^2$ via
 
-$$S^2 = \left((x_1 - \mu_x)^2 + (y_1 - \mu_y)^2\right) + \left((x_2 - \mu_x)^2 + (y_2 - \mu_y)^2\right) +\left((x_3 - \mu_x)^2 + (y_3 - \mu_y)^2\right).$$
+$$S^2 = \left((x_1 - \mu_x)^2 + (y_1 - \mu_y)^2\right) + \left((x_2 - \mu_x)^2 + (y_2 - \mu_y)^2\right) +\left((x_3 - \mu_x)^2 + (y_3 - \mu_y)^2\right) \left((x_4 - \mu_x)^2 + (y_4 - \mu_y)^2\right).$$
```

Comment:
This equation is too long for rendering on a single line (at least in my web browser)